hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandleWithChangeLog.java

```diff
@@ -75,24 +80,35 @@ public HoodieMergeHandleWithChangeLog(HoodieWriteConfig config, String instantTi
         IOUtils.getMaxMemoryPerPartitionMerge(taskContextSupplier, config));
   }
 
-  protected boolean writeUpdateRecord(HoodieRecord<T> hoodieRecord, HoodieRecord oldRecord, Option<HoodieRecord> combineRecordOp)
+  protected boolean writeUpdateRecord(HoodieRecord<T> newRecord, HoodieRecord<T> oldRecord, Option<HoodieRecord> combineRecordOpt, Schema writerSchema)
       throws IOException {
-    final boolean result = super.writeUpdateRecord(hoodieRecord, oldRecord, combineRecordOp);
+    // TODO Remove these unnecessary newInstance invocations
```

Comment:
@wzx140 please create a ticket under RFC-46 epic, and link it here like `TODO(HUDI-xxx)`